Background-Substractor-Video-Capturer/main.cpp

```diff
@@ -0,0 +1,328 @@
+#include <atomic>
+#include <cstdint>
+#include <cstdlib>
+#include <iostream>
+#include <vector>
+
+#include <opencv2/opencv.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/video/background_segm.hpp>
+
+#include <opentok.h>
+
+#include "config.h"
+#include "otk_thread.h"
+#include "renderer.h"
+
+static std::atomic<bool> g_is_connected(false);
+static otc_publisher *g_publisher = nullptr;
+static std::atomic<bool> g_is_publishing(false);
+
+struct custom_video_capturer {
+  const otc_video_capturer *video_capturer;
+  struct otc_video_capturer_callbacks video_capturer_callbacks;
+  int format;
+  int width;
+  int height;
+  int fps;
+  otk_thread_t capturer_thread;
+  std::atomic<bool> capturer_thread_exit;
+};
+
+static otk_thread_func_return_type capturer_thread_start_function(void *arg) {
+  struct custom_video_capturer *video_capturer = static_cast<struct custom_video_capturer *>(arg);
+  if (video_capturer == nullptr) {
+    otk_thread_func_return_value;
+  }
+
+  cv::VideoCapture opencv_video_capture(0);
+  if (!opencv_video_capture.isOpened()) {
+    otk_thread_func_return_value;
+  }
+
+  // The following arguments have the default values according to
+  // https://docs.opencv.org/master/de/de1/group__video__motion.html.
+  int history = 500;
+  double var_threshold = 16;
+  bool detect_shadows = true;
```

Comment:
Should these be `const`?  Alternatively, and maybe additionally, how do you feel about defining these constants in another header file.
```
#pragma once
#define DEFAULT_HISTORY 500
```